c2corg_api/views/document.py

```diff
@@ -406,3 +411,42 @@ def get_all_fields(fields, activities, field_list_type):
     # turn a list of lists [['a', 'b'], ['b', 'c'], ['d']] into a flat set
     # ['a', 'b', 'c', 'd']
     return set(sum(fields_list, []))
+
+
+@resource(path='/document/{id}/history/{lang}', cors_policy=cors_policy)
+class HistoryRouteRest(DocumentRest):
+    """Unique class for returning history of a document.
+    """
+
+    @view(validators=[validate_id, validate_lang])
+    def get(self):
+        id = self.request.validated['id']
+        lang = self.request.validated['lang']
+
+        # FIXME conditionnal permission check (when outings implemented)
+        # type = DBSession.query(Document.type).filter(Document.id == id)
+        # if type is not Outing.type:
+        #    # validate permission (authenticated + associated)
+        #    # return 403 if not correct
+
+        results = DBSession.query(
+            DocumentVersion.id,
+            HistoryMetaData.user_id,
+            User.username,
+            HistoryMetaData.comment,
+            HistoryMetaData.written_at) \
+            .filter(DocumentVersion.document_id == id) \
+            .filter(
+                HistoryMetaData.id == DocumentVersion.history_metadata_id) \
+            .filter(User.id == HistoryMetaData.user_id) \
+            .filter(DocumentVersion.culture == lang) \
+            .order_by(DocumentVersion.id) \
+            .all()
+
+        return [serialize({
```

Comment:
* Please do not return an array as top-level element, see https://github.com/c2corg/v6_api/issues/49
* In this case you don't manually have to call `serialize`. Because you only have simple data types, the default Pyramid JSON serializer will automatically convert the list to JSON.